# Eine Analyse des LIS Open Access Publikationsgeschehen anhand der API Schnittstelle des Directory of Open Access Journals

Ziel der Arbeit: über die vom DOAJ angebotene API Schnittstelle sollen Daten zu Open Access Publikationen aus dem Bereich Bibliography.Library Science.Information Resources.extrahiert und mit dem Forschungsstand verglichen werden.

Datensatzgröße: 183 Zeitschriften ; 
Anzahl verzeichnete Zeitschriften Gesamt 18.141 ;
Datum :26.8.2022 ;
Autor: Nadine Salz ;
Abschlussarbeit Zertifkatskurs Data Librarian 2022

# Gesamte Trefferliste importieren & Ansprache der API Schnittstelle

Import der benötigen Bibliotheken sowie Abfrage des Datensatzes mit größtmögliche zu ladenden Trefferzahl

urllib .request lädt die gewünschte URL 

import json importiert die json Bibliothek

In [382]:
import urllib.request
import json

Die for Loop gibt mit range (2) die Zahlen 0,1 und 2 aus.
Die zu ladenen Seiten sind aber 1 und 2, daher +1. Danach werden diese beiden Seiten mit append() zusammengeführt
der gewählte DOAJ Datensatz dazu wird dazu aus dem Netz geladen und  mit json.loads() als Dictionary ausgegeben.


In [460]:
pages = range(182)
doaj_dataset_json = []
for pagenumber in pages:
    real_pagenumber = pagenumber + 1
    
    # Ausgabe der eigentlichen Seitenzahl wäre an dieser Stelle:
    #print(real_pagenumber) 
    #base_url = "https://doaj.org/api/search/journals/*?page=" + str(real_pagenumber) + "&pageSize=100"
    
    base_url = "https://doaj.org/api/search/journals/bibjson.subject.term:/%22Bibliography.%20Library%20science.%20Information%20resources/%22?page=" + str(real_pagenumber) + "&pageSize=100"
    print(base_url)
    doaj_dataset = urllib.request.urlopen(base_url).read()
    print(pagenumber)
    doaj_dataset_json.append(json.loads(doaj_dataset))


https://doaj.org/api/search/journals/bibjson.subject.term:/%22Bibliography.%20Library%20science.%20Information%20resources/%22?page=1&pageSize=100
0
https://doaj.org/api/search/journals/bibjson.subject.term:/%22Bibliography.%20Library%20science.%20Information%20resources/%22?page=2&pageSize=100
1


Anzeige des Datensatzes zur Analyse des Aussehens mit json(): Die Methode erzeugt ein Javaskript Objekt an der Stelle 1

In [467]:
doaj_dataset_json[1]["results"]

[{'id': '94b5aafddf514923802d2372768f30b2',
  'created_date': '2021-08-26T03:21:19Z',
  'last_updated': '2022-03-21T20:00:40Z',
  'bibjson': {'boai': True,
   'eissn': '2447-0198',
   'publication_time_weeks': 12,
   'title': 'Revista Informação na Sociedade Contemporânea',
   'oa_start': 2014,
   'apc': {'has_apc': False,
    'url': 'https://periodicos.ufrn.br/informacao/polacessaberto'},
   'article': {'orcid': False,
    'i4oc_open_citations': False,
    'license_display': ['No']},
   'copyright': {'author_retains': True,
    'url': 'https://periodicos.ufrn.br/informacao/polacessaberto'},
   'deposit_policy': {'has_policy': True,
    'url': 'http://diadorim.ibict.br/handle/1/1611',
    'service': ['Diadorim']},
   'editorial': {'review_url': 'https://periodicos.ufrn.br/informacao/avaporpares',
    'board_url': 'https://periodicos.ufrn.br/informacao/about/editorialTeam',
    'review_process': ['Double blind peer review']},
   'institution': {'name': 'Universidade Federal do Rio Grand

die for Loop gibt eine Liste mit Namen all_results_list mit der eingelesenen Ergebnisstrefferliste als Datensatz aus

In [385]:
all_results_list = []

for result_page in pages:
    #print(result_page["results"])
    all_results_list += doaj_dataset_json[result_page]["results"]

len() gibt die Anzahl der Treffer dieser Liste wieder, angwendet um zu prüfen das alle Zeitschriften importiert wurden. Dies ist mit 183 der Fall

In [386]:
len(all_results_list)

183

Mit der Methode json.dumps wird das Format der erzeugten Liste geändert zu einem JSON string.

In [387]:
all_results_json = json.dumps(all_results_list)
type(all_results_json[0])

str

In [23]:
#all_results_json = json.loads(all_results_json)

Überprüfung ob das erfolgt ist mit er Methode type() . Die Angabe str bestätigt es.

In [465]:
type(all_results_json[0])

str

Anzeige der Trefferliste an der Stelle 0

In [463]:
all_results_list[0]

{'id': '0008a8877b2046b082ef902b2df8647c',
 'created_date': '2020-07-30T21:49:52Z',
 'last_updated': '2022-03-15T17:02:49Z',
 'bibjson': {'boai': True,
  'eissn': '2676-7104',
  'publication_time_weeks': 8,
  'title': 'Frontiers in Health Informatics',
  'oa_start': 2019,
  'apc': {'has_apc': False,
   'url': 'http://ijmi.ir/index.php/IJMI/about/editorialPolicies#custom-3'},
  'article': {'license_display_example_url': 'http://ijmi.ir/index.php/IJMI/about/editorialPolicies#custom-4',
   'license_display': ['Embed']},
  'copyright': {'author_retains': True,
   'url': 'http://ijmi.ir/index.php/IJMI/about/submissions#copyrightNotice'},
  'deposit_policy': {'has_policy': False},
  'editorial': {'review_url': 'http://ijmi.ir/index.php/IJMI/about/editorialPolicies#peerReviewProcess',
   'board_url': 'http://ijmi.ir/index.php/IJMI/about/editorialTeam',
   'review_process': ['Double blind peer review']},
  'institution': {'name': 'Iranian Association of Medical Informatics'},
  'other_charges'

For loop zur Ausgabe der Spalte der Schlagworte, um zu sehen das auch die gewünschten Datensätze aus der Kategorie Bibliography. Library Science. Information resources ausgegeben wurden.Dies scheint der Fall.

In [389]:
for item in all_results_list:
    print(item["bibjson"]["subject"][0]["term"])
    

Bibliography. Library science. Information resources
Museums. Collectors and collecting
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Academies and learned societies
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Communication. Mass media
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Bibliography. Library science. Information resources
Arts i

# Analyse mit Hilfe von Pandas

Import der Pandasbibliothek und lese mit pd.read_json die Trefferliste ein

In [390]:
import pandas as pd
dataframe = pd.read_json(all_results_json)


Zeige mit head() den Kopf des Dataframes
Es zeigt sich das bibjson nicht in Spalten anzeigt wird. 

In [277]:
dataframe.head()

id          created_date  \
0  0008a8877b2046b082ef902b2df8647c  2020-07-30T21:49:52Z   
1  0059b3e34a0d4e5f809b8c1096dd42f5  2019-08-03T23:42:36Z   
2  02f38a58853d4a7ea153c1d0383ba9ad  2010-11-09T14:35:09Z   
3  05847ba5e1ef4386a656a83f4b86999e  2020-01-21T18:11:41Z   
4  0599ed873dd440e8a66779bca759a448  2018-05-08T23:58:58Z   

           last_updated                                            bibjson  \
0  2022-03-15T17:02:49Z  {'boai': True, 'eissn': '2676-7104', 'publicat...   
1  2021-04-29T13:18:58Z  {'boai': True, 'eissn': '2358-0763', 'publicat...   
2  2021-04-29T13:14:05Z  {'alternative_title': 'Journal of Library and ...   
3  2021-04-29T13:19:55Z  {'alternative_title': 'Library Journal', 'boai...   
4  2021-04-29T13:13:55Z  {'alternative_title': 'Insaniyat', 'boai': Tru...   

                             admin  
0  {'seal': False, 'ticked': True}  
1  {'seal': False, 'ticked': True}  
2  {'seal': False, 'ticked': True}  
3  {'seal': False, 'ticked': True}  
4  {'seal': False, 'ticked': True}

Um die Liste in der Spalte bibjson in weitere Spalten aufzuteilen, um den Datensatz dann weiter anzusteuern,
wird mit concat und normalize ein bestimmtes Stück des gleichen Dataframes (=bibjson) in die Tabelle nebeneinder eingefügt. pd.concat() verbindet dabei Objekte miteinander und gibt es in einem Datenframe aus. 
normalize() vereinheitlich die Werte der Zeile für eine saubere Ausgabe.
Mit axis =1 werden die beiden Datensätze nicht untereinander zusammengefügt sondern nebeneinander. 
Die Spalten vom 2. Datensatz erscheinen dadurch neben dem 1. Datensatz.
drop()entfernt die ursprüngliche bibjson Spalte.

In [391]:
from pandas import json_normalize

df = pd.concat([pd.DataFrame(dataframe), 
                json_normalize(dataframe['bibjson'])], 
                axis=1).drop('bibjson', 1)

/tmp/ipykernel_102839/397273336.py:3: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  df = pd.concat([pd.DataFrame(dataframe),


df gibt das bearbeitete Datenframe aus

In [279]:
df

id          created_date  \
0    0008a8877b2046b082ef902b2df8647c  2020-07-30T21:49:52Z   
1    0059b3e34a0d4e5f809b8c1096dd42f5  2019-08-03T23:42:36Z   
2    02f38a58853d4a7ea153c1d0383ba9ad  2010-11-09T14:35:09Z   
3    05847ba5e1ef4386a656a83f4b86999e  2020-01-21T18:11:41Z   
4    0599ed873dd440e8a66779bca759a448  2018-05-08T23:58:58Z   
..                                ...                   ...   
178  f73dfa7769bf417b900143ca4d8951a1  2013-06-23T18:20:28Z   
179  f7f587617c134cd99447c6ff8aa6c64a  2006-12-21T14:32:17Z   
180  f936908299e446c1b5c9d8c108cc596f  2019-10-20T03:40:44Z   
181  fb979e39ff8444eaa7aed77df8ed9169  2017-07-03T13:16:11Z   
182  fd40919b9baf458f8cff4d11a1f294eb  2019-03-06T16:29:15Z   

             last_updated                            admin  boai      eissn  \
0    2022-03-15T17:02:49Z  {'seal': False, 'ticked': True}  True  2676-7104   
1    2021-04-29T13:18:58Z  {'seal': False, 'ticked': True}  True  2358-0763   
2    2021-04-29T13:14:05Z  {'seal': False, 'ticked': True}  True  2224-1574   
3    2021-04-29T13:19:55Z  {'seal': False, 'ticked': True}  True        NaN   
4    2021-04-29T13:13:55Z  {'seal': False, 'ticked': True}  True  2541-500X   
..                    ...                              ...   ...        ...   
178  2021-04-29T13:14:36Z  {'seal': False, 'ticked': True}  True  2287-187X   
179  2021-08-16T15:35:08Z  {'seal': False, 'ticked': True}  True  1746-8256   
180  2021-04-29T13:17:58Z  {'seal': False, 'ticked': True}  True  2238-5436   
181  2021-04-29T13:18:35Z  {'seal': False, 'ticked': True}  True  2477-0361   
182  2021-04-29T13:19:46Z   {'seal': True, 'ticked': True}  True  2398-4112   

     publication_time_weeks  \
0                         8   
1                        15   
2                        24   
3                         7   
4                        16   
..                      ...   
178                       5   
179                      30   
180                      22   
181                      12   
182                      30   

                                                 title  oa_start  \
0                      Frontiers in Health Informatics      2019   
1                     Ciência da Informação em Revista      2014   
2                          Tushuguanxue yu Zixun Kexue      1975   
3                                  Bìblìotečnij vìsnik      2012   
4           Insaniyat: Journal of Islam and Humanities      2016   
..                                                 ...       ...   
178  International Journal of Knowledge Content Dev...      2011   
179          International Journal of Digital Curation      2006   
180         Revista Museologia & Interdisciplinaridade      2012   
181            Berkala Ilmu Perpustakaan dan Informasi      2003   
182                                               KULA      2017   

                                              keywords  ...  \
0    [medical informatics, image processing, decisi...  ...   
1    [information science, library science, librari...  ...   
2    [information literacy, information behaviour, ...  ...   
3    [library and information sciences, social comm...  ...   
4    [islam, humanities, history, literature, libra...  ...   
..                                                 ...  ...   
178  [information science, content development, web...  ...   
179  [digital preservation, digital access, researc...  ...   
180                                        [museology]  ...   
181  [literacy and reading, libraries, information ...  ...   
182  [human knowledge processes, knowledge creation...  ...   

          preservation.service  \
0                          NaN   
1          [Cariniana Network]   
2                          NaN   
3                          NaN   
4                          NaN   
..                         ...   
178                        NaN   
179                        NaN   
180        [Cariniana Network]   
181                        NaN   
182  [CL

list() erzeugt eine Liste der Spalten. Ziel ist zu schauen, wie die Bezeichnung der Objekte lautet von denen, die betrachtet werden sollen.

In [392]:
list(df.columns)

['id',
 'created_date',
 'last_updated',
 'admin',
 'boai',
 'eissn',
 'publication_time_weeks',
 'title',
 'oa_start',
 'keywords',
 'language',
 'license',
 'subject',
 'apc.has_apc',
 'apc.url',
 'article.license_display_example_url',
 'article.license_display',
 'copyright.author_retains',
 'copyright.url',
 'deposit_policy.has_policy',
 'editorial.review_url',
 'editorial.board_url',
 'editorial.review_process',
 'institution.name',
 'other_charges.has_other_charges',
 'pid_scheme.has_pid_scheme',
 'pid_scheme.scheme',
 'plagiarism.detection',
 'plagiarism.url',
 'preservation.has_preservation',
 'publisher.name',
 'publisher.country',
 'ref.oa_statement',
 'ref.journal',
 'ref.aims_scope',
 'ref.author_instructions',
 'ref.license_terms',
 'waiver.has_waiver',
 'article.orcid',
 'article.i4oc_open_citations',
 'deposit_policy.service',
 'preservation.url',
 'preservation.service',
 'alternative_title',
 'pissn',
 'other_charges.url',
 'preservation.national_library',
 'deposit_po

Anzeige der zu betrachtenden Spalten mit:

In [393]:
df[["id", "oa_start","language" , "license", "editorial.review_process" , "apc.has_apc" , "publisher.country" , "publisher.name" ]]

id  oa_start  language  \
0    0008a8877b2046b082ef902b2df8647c      2019      [EN]   
1    0059b3e34a0d4e5f809b8c1096dd42f5      2014      [PT]   
2    02f38a58853d4a7ea153c1d0383ba9ad      1975  [ZH, EN]   
3    05847ba5e1ef4386a656a83f4b86999e      2012  [UK, EN]   
4    0599ed873dd440e8a66779bca759a448      2016  [AR, EN]   
..                                ...       ...       ...   
178  f73dfa7769bf417b900143ca4d8951a1      2011      [EN]   
179  f7f587617c134cd99447c6ff8aa6c64a      2006      [EN]   
180  f936908299e446c1b5c9d8c108cc596f      2012      [PT]   
181  fb979e39ff8444eaa7aed77df8ed9169      2003  [EN, ID]   
182  fd40919b9baf458f8cff4d11a1f294eb      2017      [EN]   

                                               license  \
0    [{'type': 'CC BY-NC', 'BY': True, 'NC': True, ...   
1    [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...   
2    [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...   
3    [{'type': 'CC BY-NC', 'BY': True, 'NC': True, ...   
4    [{'type': 'CC BY-NC-SA', 'BY': True, 'NC': Tru...   
..                                                 ...   
178  [{'type': 'CC BY-NC-ND', 'BY': True, 'NC': Tru...   
179  [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...   
180  [{'type': 'CC BY-NC-ND', 'BY': True, 'NC': Tru...   
181  [{'type': 'CC BY-SA', 'BY': True, 'NC': False,...   
182  [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...   

                         editorial.review_process  apc.has_apc  \
0                      [Double blind peer review]        False   
1                      [Double blind peer review]        False   
2                      [Double blind peer review]        False   
3                      [Double blind peer review]        False   
4                             [Blind peer review]        False   
..                                            ...          ...   
178                                 [Peer review]        False   
179                    [Double blind peer review]        False   
180                    [Double blind peer review]        False   
181                    [Double blind peer review]        False   
182  [Editorial review, Double blind peer review]        False   

    publisher.country                                     publisher.name  
0                  IR                                       Hamara Afzar  
1                  BR                    Universidade Federal de Alagoas  
2                  TW                  National Taiwan Normal University  
3                  UA              Vernadsky National Library of Ukraine  
4                  ID  Faculty of Adab and Humanities UIN Syarif Hida...  
..                ...                                                ...  
178                KR  Research Institute for Knowledge Content Devel...  
179                GB                            University of Edinburgh  
180                BR                           Universidade de Brasília  
181                ID                            Universitas Gadjah Mada  
182                CA                   University of Victoria Libraries  

[183 rows x 8 columns]

# Bereinigung von NAN Werten

Prüfen, ob in den zu betrachteten Spalten NAN Werte vorkommen, um sie ggf zu füllen oder löschen.

mit copy() Kopie erstellt für den Fall, dass Werte gelöscht oder mit unknown gefüllt hätten müssen, um den ursprünglichen Datensatz zur Sicherheit zu erhalten. Da keine NAN Werte vorhanden sind, kann dieser Schritt ausgelassen werden.

In [394]:
df_Kopie = df.copy()

isnull().sum() Summiere die Anzahl der Zeilen, wo keine NAN Werte angegeben werden. Der ISSN sind also 9 Zeilen mit NAN Werten zu finden.

In [395]:
df_Kopie.isnull().sum()

id                                       0
created_date                             0
last_updated                             0
admin                                    0
boai                                     0
eissn                                    9
publication_time_weeks                   0
title                                    0
oa_start                                 0
keywords                                 0
language                                 0
license                                  0
subject                                  0
apc.has_apc                              0
apc.url                                  0
article.license_display_example_url     82
article.license_display                  0
copyright.author_retains                 0
copyright.url                           33
deposit_policy.has_policy                0
editorial.review_url                     0
editorial.board_url                      0
editorial.review_process                 0
institution

# Dublettenprüfung

Überprüfung zur Sicherheit ob es ggf. Dubletten geben könnte, die rausgezogen werden müssten. Ist nicht der Fall

dublicated(subset) übergibt angegebene Spalte, in dem Fall id zur Dublettenprüfung. ID gewählt, da die einzigartig ist und eine gesicherte Aussage dazu gibt ob eine Dublette enthalten ist. Es ist keine Dublette vorhanden.

In [311]:
df_Kopie_Dubletten = df_Kopie[df_Kopie.duplicated(subset = "id")]
df_Kopie_Dubletten

Empty DataFrame
Columns: [id, created_date, last_updated, admin, boai, eissn, publication_time_weeks, title, oa_start, keywords, language, license, subject, apc.has_apc, apc.url, article.license_display_example_url, article.license_display, copyright.author_retains, copyright.url, deposit_policy.has_policy, editorial.review_url, editorial.board_url, editorial.review_process, institution.name, other_charges.has_other_charges, pid_scheme.has_pid_scheme, pid_scheme.scheme, plagiarism.detection, plagiarism.url, preservation.has_preservation, publisher.name, publisher.country, ref.oa_statement, ref.journal, ref.aims_scope, ref.author_instructions, ref.license_terms, waiver.has_waiver, article.orcid, article.i4oc_open_citations, deposit_policy.service, preservation.url, preservation.service, alternative_title, pissn, other_charges.url, preservation.national_library, deposit_policy.url, institution.country, apc.max, waiver.url, replaces]
Index: []

[0 rows x 52 columns]

# Start der Datenanalyse und Beantwortung der Fragen

# Betrachtung OA Start: Wann ist der erste und wann der letzte OA Start?

.loc gibt Datenindexwert an. Zeigt aus den "Spalten" 0-183 die Werte aus der Spalte Open Access start an. Die Liste ist noch unsortiert.

In [142]:
df_Kopie.loc[:, ["oa_start"]]

oa_start
0        2019
1        2014
2        1975
3        2012
4        2016
..        ...
178      2011
179      2006
180      2012
181      2003
182      2017

[183 rows x 1 columns]

sort_values sortiert die Werte der Spalte Open Access start

In [461]:
df_Kopie.sort_values(by=["oa_start"])

id          created_date  \
177  f69de99a68644f7b8aa73b2343cc28a4  2013-10-29T08:51:00Z   
171  f24d1935251c4802993ef202f1124ede  2006-11-06T14:49:32Z   
2    02f38a58853d4a7ea153c1d0383ba9ad  2010-11-09T14:35:09Z   
142  d47bf74be3c44b849d6f73f738b71a52  2015-05-21T11:05:32Z   
92   8d442f2ade5c41919bf471bbdd7b28d0  2017-09-26T22:33:21Z   
..                                ...                   ...   
0    0008a8877b2046b082ef902b2df8647c  2020-07-30T21:49:52Z   
123  af3864c617b346578fbc410831f8abb5  2020-10-12T07:42:20Z   
145  d8fee705be2949e2a84e7cb6d97b87f9  2021-12-15T17:10:01Z   
21   1a7136a04b8d482da3a650b08d0e5596  2022-07-06T15:02:25Z   
6    07a8b9e848a7462c91952373e9a10313  2020-07-15T12:57:16Z   

             last_updated                            admin  boai      eissn  \
177  2021-08-24T14:27:46Z  {'seal': False, 'ticked': True}  True  2150-6701   
171  2021-04-29T13:14:02Z  {'seal': False, 'ticked': True}  True        NaN   
2    2021-04-29T13:14:05Z  {'seal': False, 'ticked': True}  True  2224-1574   
142  2021-04-29T13:18:43Z  {'seal': False, 'ticked': True}  True  1988-2890   
92   2021-04-29T13:17:18Z  {'seal': False, 'ticked': True}  True  1659-3286   
..                    ...                              ...   ...        ...   
0    2022-03-15T17:02:49Z  {'seal': False, 'ticked': True}  True  2676-7104   
123  2021-04-29T13:19:50Z  {'seal': False, 'ticked': True}  True  2689-5870   
145  2021-12-15T17:10:01Z  {'seal': False, 'ticked': True}  True  2240-3604   
21   2022-07-06T15:02:25Z  {'seal': False, 'ticked': True}  True  2746-0509   
6    2021-04-29T13:14:29Z  {'seal': False, 'ticked': True}  True  2518-3354   

     publication_time_weeks                                            title  \
177                      45                   College and Research Libraries   
171                      12                         North Carolina Libraries   
2                        24                      Tushuguanxue yu Zixun Kexue   
142                       6  Documentación de las Ciencias de la Información   
92                       20                              Revista Bibliotecas   
..                      ...                                              ...   
0                         8                  Frontiers in Health Informatics   
123                      12                     Scholarly Assessment Reports   
145                      12                                             TECA   
21                        4                                        Al-Ma'mun   
6                        12                         European Science Editing   

     oa_start                                           keywords  ...  \
177      1939  [library science, information literacy, higher...  ...   
171      1942  [librarianship, media centers, academic librar...  ...   
2        1975  [information literacy, information behaviour, ...  ...   
142      1976  [communication, audiovisual, media, librarian ...  ...   
92       1980                [librarianship, information, users]  ...   
..        ...                                                ...  ...   
0        2019  [medical informatics, image processing, decisi...  ...   
123      2020  [metascience, infometrics, bibliometrics, asse...  ...   
145      2020  [history of libraries, history of the book, li...  ...   
21       2020  [digital library, library and information, lib...  ...   
6        2020  [editing, publishing, academia, scientific pub...  ...   

    preservation.service                                  alternative_title  \
177    [LOCKSS, Portico]                                                NaN   
171                  NaN                                                NaN   
2                    NaN         Journal of Library and Information Science   
142                  NaN                                                NaN   
92             [PKP PLN]  Revista de la Escuela de Bibliotecología, Docu...   
..               

# Antwort: Der OA Start reicht von 1939 bis 2020

# In welchem Jahr wurden die meisten OA Starts registriert?

groupby() und count() Gruppieren und zählen die häufigkeit der Werte der Spalte Open Access und zeigt an, wieviele Zeitschriften im Jahr 1993 zu OA Journals wurden.

In [152]:
df_Kopie.groupby("oa_start").oa_start.count()

oa_start
1939     1
1942     1
1975     1
1976     1
1980     1
1981     1
1986     2
1987     1
1989     1
1992     1
1993     1
1995     3
1996     2
1997     1
1998     4
1999     4
2000     1
2001     1
2002     2
2003     6
2004     5
2005     4
2006     5
2007     9
2008    11
2009     5
2010     4
2011    14
2012    10
2013    12
2014    13
2015    10
2016    15
2017     7
2018    13
2019     6
2020     4
Name: oa_start, dtype: int64

# Antwort: 2016 wurden die meisten Zeitschriften als OA Journals herausgegeben.

Mit dtypes können die Art der Objekte angezeigt werden. Wenn Fehlerabfragen darauf hinweisen, dass der falsche Typ verwendet wird.Da dies Häufig passiert ist, wird es an dieser Stelle geprüft.

In [167]:
df_Kopie.dtypes

id                                     object
created_date                           object
last_updated                           object
admin                                  object
boai                                     bool
eissn                                  object
publication_time_weeks                  int64
title                                  object
oa_start                                int64
keywords                               object
language                               object
license                                object
subject                                object
apc.has_apc                              bool
apc.url                                object
article.license_display_example_url    object
article.license_display                object
copyright.author_retains                 bool
copyright.url                          object
deposit_policy.has_policy                bool
editorial.review_url                   object
editorial.board_url               

# Welche der OA Lizenzen werden vornehmlich vergeben?

loc() Zeigt den Inhalt von der "Spalte" Lizenz an. Es wird deutlich dass die Daten verschachtelt sind. So kann keine Zählung vorgenommen werden.

In [344]:
df_Kopie.loc[:, ["license"]]

license
0    [{'type': 'CC BY-NC', 'BY': True, 'NC': True, ...
1    [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...
2    [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...
3    [{'type': 'CC BY-NC', 'BY': True, 'NC': True, ...
4    [{'type': 'CC BY-NC-SA', 'BY': True, 'NC': Tru...
..                                                 ...
178  [{'type': 'CC BY-NC-ND', 'BY': True, 'NC': Tru...
179  [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...
180  [{'type': 'CC BY-NC-ND', 'BY': True, 'NC': Tru...
181  [{'type': 'CC BY-SA', 'BY': True, 'NC': False,...
182  [{'type': 'CC BY', 'BY': True, 'NC': False, 'N...

[183 rows x 1 columns]

mit der for loop und append sowie value_counts() wird eine Liste erzeugt, die die gewählten Objekte in die gewünschte Form bringt. Wichtig dabei ist der genaue "Ort": bibjson license 0 type" entspricht daher der Struktur die in der schriftlichen Ausarbeitung beigefügt ist.

In [471]:
list_license = []
for item in all_results_list :
    list_license.append(item["bibjson"]["license"][0]["type"])
dataframe_license = pd.DataFrame(list_license)
dataframe_license.value_counts()

CC BY                      81
CC BY-NC                   36
CC BY-NC-ND                31
CC BY-NC-SA                17
CC BY-SA                   13
CC BY-ND                    3
Publisher's own license     2
dtype: int64

# Antwort: Die am Häufigsten vergebenen Lizenzen sind CC BY Lizenzen

groupby(level=0).max() Groupiert Daten nach Größe nach Angabe 

id          created_date  \
0    0008a8877b2046b082ef902b2df8647c  2020-07-30T21:49:52Z   
1    0059b3e34a0d4e5f809b8c1096dd42f5  2019-08-03T23:42:36Z   
2    02f38a58853d4a7ea153c1d0383ba9ad  2010-11-09T14:35:09Z   
3    05847ba5e1ef4386a656a83f4b86999e  2020-01-21T18:11:41Z   
4    0599ed873dd440e8a66779bca759a448  2018-05-08T23:58:58Z   
..                                ...                   ...   
178  f73dfa7769bf417b900143ca4d8951a1  2013-06-23T18:20:28Z   
179  f7f587617c134cd99447c6ff8aa6c64a  2006-12-21T14:32:17Z   
180  f936908299e446c1b5c9d8c108cc596f  2019-10-20T03:40:44Z   
181  fb979e39ff8444eaa7aed77df8ed9169  2017-07-03T13:16:11Z   
182  fd40919b9baf458f8cff4d11a1f294eb  2019-03-06T16:29:15Z   

             last_updated                            admin  boai      eissn  \
0    2022-03-15T17:02:49Z  {'seal': False, 'ticked': True}  True  2676-7104   
1    2021-04-29T13:18:58Z  {'seal': False, 'ticked': True}  True  2358-0763   
2    2021-04-29T13:14:05Z  {'seal': False, 'ticked': True}  True  2224-1574   
3    2021-04-29T13:19:55Z  {'seal': False, 'ticked': True}  True        NaN   
4    2021-04-29T13:13:55Z  {'seal': False, 'ticked': True}  True  2541-500X   
..                    ...                              ...   ...        ...   
178  2021-04-29T13:14:36Z  {'seal': False, 'ticked': True}  True  2287-187X   
179  2021-08-16T15:35:08Z  {'seal': False, 'ticked': True}  True  1746-8256   
180  2021-04-29T13:17:58Z  {'seal': False, 'ticked': True}  True  2238-5436   
181  2021-04-29T13:18:35Z  {'seal': False, 'ticked': True}  True  2477-0361   
182  2021-04-29T13:19:46Z   {'seal': True, 'ticked': True}  True  2398-4112   

     publication_time_weeks  \
0                         8   
1                        15   
2                        24   
3                         7   
4                        16   
..                      ...   
178                       5   
179                      30   
180                      22   
181                      12   
182                      30   

                                                 title  oa_start  \
0                      Frontiers in Health Informatics      2019   
1                     Ciência da Informação em Revista      2014   
2                          Tushuguanxue yu Zixun Kexue      1975   
3                                  Bìblìotečnij vìsnik      2012   
4           Insaniyat: Journal of Islam and Humanities      2016   
..                                                 ...       ...   
178  International Journal of Knowledge Content Dev...      2011   
179          International Journal of Digital Curation      2006   
180         Revista Museologia & Interdisciplinaridade      2012   
181            Berkala Ilmu Perpustakaan dan Informasi      2003   
182                                               KULA      2017   

                                              keywords  ...  \
0    [medical informatics, image processing, decisi...  ...   
1    [information science, library science, librari...  ...   
2    [information literacy, information behaviour, ...  ...   
3    [library and information sciences, social comm...  ...   
4    [islam, humanities, history, literature, libra...  ...   
..                                                 ...  ...   
178  [information science, content development, web...  ...   
179  [digital preservation, digital access, researc...  ...   
180                                        [museology]  ...   
181  [literacy and reading, libraries, information ...  ...   
182  [human knowledge processes, knowledge creation...  ...   

          preservation.service  \
0                          NaN   
1          [Cariniana Network]   
2                          NaN   
3                          NaN   
4                          NaN   
..                         ...   
178                        NaN   
179                        NaN   
180        [Cariniana Network]   
181                        NaN   
182  [CL

Mit .loc Anzeige des der Länder in einer Liste

In [475]:
df_Kopie.loc[:, ["publisher.country"]]

publisher.country
0                  IR
1                  BR
2                  TW
3                  UA
4                  ID
..                ...
178                KR
179                GB
180                BR
181                ID
182                CA

[183 rows x 1 columns]

Mit groupby().size().sort_values(ascending=False) wird die absteigende Sortierung der Häufigkeiten wiedergegeben.

In [476]:
ser_grouped_publisher_country = df.groupby("publisher.country").size().sort_values(ascending=False)
ser_grouped_publisher_country 

publisher.country
US    29
BR    24
ID    13
ES    12
PL     9
IR     9
UA     7
IT     6
GB     6
CA     6
DE     5
CH     5
RO     4
EG     3
MX     3
CR     2
RU     2
NL     2
NG     2
LT     2
KR     2
AR     2
CU     2
FR     2
TW     2
CN     2
SE     1
SG     1
TR     1
PT     1
RS     1
QA     1
UY     1
KE     1
PK     1
NO     1
MY     1
AT     1
JP     1
IN     1
HR     1
FI     1
CO     1
BG     1
BA     1
ZA     1
dtype: int64

# Antwort : USA 29 ; BR: 24 ; ID: 13 ; ES: 12 GB: 6 DE:5

In [ ]:
mit .loc() und Auswahl des Länderkürzels kann auch eine Einzelabfrage nach gezielten Ländern getätigt werden.

In [199]:
ser_grouped_publisher_country.loc["US"]

29

In [177]:
ser_grouped_publisher_country.loc["GB"]

6

# Wieviele sind peer-reviewed und double blind peer-reviewed?

Mit .loc() Anzeige des Spalteninhalts von editorial.review_process. Auch dieser Auszug zeigt, das die Einträge nicht einzeln sind (siehe Treffer 182)

In [404]:
df_Kopie.loc[:, ["editorial.review_process"]]

editorial.review_process
0                      [Double blind peer review]
1                      [Double blind peer review]
2                      [Double blind peer review]
3                      [Double blind peer review]
4                             [Blind peer review]
..                                            ...
178                                 [Peer review]
179                    [Double blind peer review]
180                    [Double blind peer review]
181                    [Double blind peer review]
182  [Editorial review, Double blind peer review]

[183 rows x 1 columns]

Daher wieder mit for Loop wie bereits oben Beschrieben mit append() und value_counts die Anzahl der gewünschten einzelnen Einträge an der "Stelle" review_process" innerhalb des dictionaries ermitteln.

In [437]:
list_review_process = []
for item in all_results_list :
    list_review_process.append(item["bibjson"]["editorial"]["review_process"][0])
dataframe_review_process = pd.DataFrame(list_review_process )
dataframe_review_process.value_counts()

Double blind peer review    114
Blind peer review            31
Peer review                  28
Editorial review              6
Open peer review              3
Committee review              1
dtype: int64

# Antwort: Double blind peer review kommt am Häufigsten vor und deutet damit auf eine Hohe Qualität der gelisteten OA Journals hin.

# Wieviele verlangen Publikationsgebühren?

Mit .loc Alle Zeilen der Spalte apc.has_apc in eine Liste übergeben

In [439]:
df_Kopie.loc[:, ["apc.has_apc"]]

apc.has_apc
0          False
1          False
2          False
3          False
4          False
..           ...
178        False
179        False
180        False
181        False
182        False

[183 rows x 1 columns]

Wieder mit groupby().size() die Häufigkeit errechnen:

In [398]:
ser_grouped_apc = df.groupby("apc.has_apc").size()
ser_grouped_apc

apc.has_apc
False    170
True      13
dtype: int64

# Antwort: Von 183 verlangen nur 13 Publisher APC Gebühren. 170 dagegen keine.

# Sind unter den Herausgebern namenhafte wie de Gruyter?

In [490]:
list_publisher_name = []
for item in all_results_list:
    list_publisher_name.append(item["bibjson"]["publisher"]["name"])
dataframe_language = pd.DataFrame(list_publisher_name)
dataframe_language

0
0                                         Hamara Afzar
1                      Universidade Federal de Alagoas
2                    National Taiwan Normal University
3                Vernadsky National Library of Ukraine
4    Faculty of Adab and Humanities UIN Syarif Hida...
..                                                 ...
178  Research Institute for Knowledge Content Devel...
179                            University of Edinburgh
180                           Universidade de Brasília
181                            Universitas Gadjah Mada
182                   University of Victoria Libraries

[183 rows x 1 columns]

In [498]:
ser_grouped_list_publisher_name.loc["Hamara Afzar"]

1

In [502]:
ser_grouped_list_publisher_name.loc["University of Edinburgh"]

1

In [492]:
ser_grouped_list_publisher_name = df.groupby("publisher.name").size().sort_values(ascending=False)
ser_grouped_list_publisher_name 

publisher.name
University Library System, University of Pittsburgh       3
Universidad Complutense de Madrid                         3
Levy Library Press                                        2
University of Alberta                                     2
Universidad de Murcia                                     2
                                                         ..
Library Association of the City University of New York    1
Library of the University of Heidelberg                   1
Lodz University Press                                     1
Marketing Libraries Journal                               1
openjournals.nl                                           1
Length: 165, dtype: int64

 Leider funktioniert keine .query Abfrage oder ähnliches. Ich gehe davon aus in der Falschen "Spalte zu sein. Es sind auch nur noch 165 Datensätze. Anhand der Schnittstellensuche in der API Testsuche konnte ermittelt werden, dass es lediglich ein OA Journal von de Gruyter enthält.